web/packages/shared/components/UnifiedResources/shared/PinButton.tsx

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright 2023 Gravitational, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useRef } from 'react';
+
+import { Box } from 'design';
+import { PushPinFilled, PushPin } from 'design/Icon';
+import ButtonIcon from 'design/ButtonIcon';
+
+import { HoverTooltip } from 'shared/components/ToolTip';
+
+import { PinningSupport } from '../types';
+
+import { PINNING_NOT_SUPPORTED_MESSAGE } from '../UnifiedResources';
+
+export function PinButton({
+  pinned,
+  pinningSupport,
+  hovered,
+  setPinned,
+  className,
+}: {
+  pinned: boolean;
+  pinningSupport: PinningSupport;
+  hovered: boolean;
+  setPinned: () => void;
+  className?: string;
+}) {
+  const copyAnchorEl = useRef(null);
+  const tipContent = getTipContent(pinningSupport, pinned);
+
+  const shouldShowButton =
+    pinningSupport !== PinningSupport.Hidden && (pinned || hovered);
+  const shouldDisableButton =
+    pinningSupport === PinningSupport.Disabled ||
+    pinningSupport === PinningSupport.NotSupported;
+
+  const $content = pinned ? (
+    <PushPinFilled color="brand" size="small" />
+  ) : (
+    <PushPin size="small" />
+  );
+
+  return (
+    // This is wrapped in a Box so that we can disable the transition for showing/hiding the button while not
+    // interfering with the ButtonIcon's transitions.
+    <Box
+      css={`
+        // dont display but keep the layout
+        opacity: ${shouldShowButton ? '1' : '0'};
```

Comment:
I think we can go back to 
```ts
visibility: ${shouldShowButton ? 'visible' : 'hidden'};
```
Now when the `PinButton` is hidden, hovering over it changes the cursor to pointer.

Reply:
I had done this because a `transition` on `visibility` just delays it, which means that when you hovered off of a row the pin button wouldn't disappear instantly. I've found a workaround that works better now